apps/aecore/test/aecore_channels_compatibility_test.exs

```diff
@@ -197,28 +199,217 @@ defmodule AecoreChannelCompatibilityTest do
     assert_txs_equal(tx, epoch_settle_tx)
   end
 
+  @tag :channels
+  @tag :compatibility
+  test "Full compatibility test (create, 2xtransfer, mutal close)", ctx do
+    # create
+    initiator_amount = 100
+    responder_amount = 200
+    channel_reserve = 40
+    initiator_nonce = 1
+    fee = 1
+    locktime = 6
+
+    tmp_id = <<1, 2, 3, 4, 5>>
+
+    assert :ok ==
+             call_s1(
+               {:initialize, tmp_id, ctx.pk_initiator, ctx.pk_responder, :initiator,
+                channel_reserve}
+             )
+
+    assert :ok ==
+             call_s2(
+               {:initialize, tmp_id, ctx.pk_initiator, ctx.pk_responder, :responder,
+                channel_reserve}
+             )
+
+    {:ok, id, half_open_tx} =
+      call_s1(
+        {:open, tmp_id, initiator_amount, responder_amount, locktime, fee, initiator_nonce,
+         ctx.sk_initiator}
+      )
+
+    {:ok, ^id, open_tx} =
+      call_s2(
+        {:sign_open, tmp_id, initiator_amount, responder_amount, locktime, half_open_tx,
+         ctx.sk_responder}
+      )
+
+    epoch_channel =
+      <<248, 111, 58, 1, 161, 1, 195, 127, 140, 188, 222, 21, 148, 121, 3, 245, 220, 105, 162,
+        143, 84, 114, 8, 161, 100, 45, 92, 39, 172, 108, 6, 12, 3, 120, 185, 238, 238, 133, 161,
+        1, 246, 50, 15, 95, 253, 247, 7, 8, 114, 192, 202, 92, 31, 249, 69, 161, 170, 113, 41, 30,
+        168, 250, 11, 241, 209, 7, 58, 85, 192, 148, 250, 1, 130, 1, 44, 100, 40, 160, 197, 119,
+        126, 105, 194, 1, 235, 15, 248, 31, 86, 174, 29, 186, 242, 64, 110, 152, 19, 149, 123,
+        111, 63, 231, 79, 148, 218, 78, 165, 165, 37, 21, 0, 6, 0>>
+
+    {:ok, %Chainstate{channels: channels}} =
+      Chainstate.calculate_and_validate_chain_state(
+        [open_tx],
+        ctx.chainstate,
+        ctx.next_height,
+        ctx.pk_initiator
+      )
+
+    our_channel = ChannelStateTree.get(channels, id)
+
+    assert_channels_equal(our_channel, epoch_channel)
+
+    assert :ok == call_s1({:receive_confirmed_tx, open_tx})
+    assert :ok == call_s2({:receive_confirmed_tx, open_tx})
+
+    # transfer from responder to initiator with amount 70
+    epoch_transfer1 =
+      Bits.decode58(
+        "G6DYbwbT9jsa4M7dKjQAw1bSwcPN3TcaCKtxBBAk28QaXfEugae4HGVNxSHSrSd59vPCLAssWmX8vN48LWwTypFq2FYy8AVAev68Q5WerLhcsCViNAtgNdzdeGhRMYHBHxYwnPdMRjKThcQAogdqjrtnMKnwqVE6awwDPYPmRjsDT3JW8HY8m43SWB9QeiauG3EZx8KnjejyXuNZNPGVjFiGnjQCAhwVb5nPXuhs4KXgTKjB5w1Huch5wJg3HUzyQXhuoCk4gn7vcQvjX7P4rthpo1CMNfJj6WXFowKpVzUCQrVrTYUMHzvbYuymzh1CdcmfgJouxHQmDSHXJwRW6Uw6uwRY192oKMbhsanuLjRJC9YB8SzSLdrSu828c6QLHnWqPedsHwW"
```

Comment:
I assume most of the hardcoded things will break with all updates to the protocol?